docs/examples/efficientnet/model_repository/preprocessing/1/dali.py

```diff
@@ -0,0 +1,39 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2023 NVIDIA CORPORATION
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy of
+# this software and associated documentation files (the "Software"), to deal in
+# the Software without restriction, including without limitation the rights to
+# use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+# the Software, and to permit persons to whom the Software is furnished to do so,
+# subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+# FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+# COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+# IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+# CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+import nvidia.dali as dali
+import nvidia.dali.types as types
+from nvidia.dali.plugin.triton import autoserialize
+
+
+@autoserialize
+@dali.pipeline_def(batch_size=512, num_threads=1, device_id=0)
```

Comment:
Shouldn't the batch size be updated as well?
Maybe you can expose `hw_decoder_load` and `num_threads` as one of the tuning knobs from the main script?

Reply:
> Shouldn't the batch size be updated as well?

The batch size set in the config overrides the one specified here. Therefore it is not necessary to update this one, as in the setup stage the batch size is always injected into the config.

I wouldn't like to expose `hw_decoder_load` and `num_threads`. The reason is that there may be more parameters to tune and nobs to turn and if we start exposing those that are unnecessary in the setup stage, we will need to answer more questions: "Why not exposing number of DALI instances?", "Why not exposing number of Efficientnet instances" etc... The batch_size is exposed only because it is necessary for the conversion. And the intention of the script is for the user to modify any other use-case-specific parameter.

Reply:
Sure